addons/snailmail_account/wizard/account_invoice_send.py

```diff
@@ -21,7 +21,8 @@ class AccountInvoiceSend(models.TransientModel):
     @api.depends('invoice_ids')
     def _compute_invalid_addresses(self):
         for wizard in self:
-            invalid_invoices = wizard.invoice_ids.filtered(lambda i: not self.env['snailmail.letter']._is_valid_address(i.partner_id))
+            invalid_invoices = wizard.invoice_ids.filtered(lambda i: not i.partner_id
+                                                           or not self.env['snailmail.letter']._is_valid_address(i.partner_id))
```

Comment:
Isn't this a bugfix that should be done in a stable version?

___________

You indentation is weird; I see a few options here
```python
            invalid_invoices = wizard.invoice_ids.filtered(
                lambda i: not i.partner_id or not self.env['snailmail.letter']._is_valid_address(i.partner_id)
            )
```
```python
            invalid_invoices = wizard.invoice_ids.filtered(lambda i: (
                not i.partner_id
                or not self.env['snailmail.letter']._is_valid_address(i.partner_id)
            ))
```
```python
            invalid_invoices = wizard.invoice_ids.filtered(
                lambda i: (
                    not i.partner_id
                    or not self.env['snailmail.letter']._is_valid_address(i.partner_id)
                )
            )
```
and even
```python
            invalid_invoices = wizard.invoice_ids.filtered(
                lambda i: not (i.partner_id and self.env['snailmail.letter']._is_valid_address(i.partner_id))
            ))
```
____________

But I would say that we should support having an empty recordset in `_is_valid_address`

```python
    @api.model
    def _is_valid_address(self, record):
        required_keys = ['street', 'city', 'zip', 'country_id']
        return len(record) <= 1 and all(record[key] for key in required_keys)
```

Reply:
Indeed sorry. I didn't even think of checking the stable versions, shame on me.
It seems like the issue goes back to v13, so I'll make another task/branch to fix it as you said by allowing empty recordsets in _is_valid_address().